BifCl.cmake needs Zeek::BifCl triggered when touching CMakeLists.txt

When developing a plugin with a `./configure && make` workflow, in Zeek 5.2, after modifying/touching CMakeLists.txt (e.g adding more sources or changing flags), then re-running `make` would trigger a cmake re-run and continue the build usually succeeding. With latest master, modifying the CMakeLists.txt now hard-fails the re-run of cmake:

In a zeek-community-id checkout:
```
$ ./configure
$ cd build
$ make    # success
$ touch ../CMakeLists.txt
$ make    # success expected
```

With Zeek 6.0-rc1, the last step fails as follows:
```
$ make
CMake Error at /opt/zeek-dev/share/zeek/cmake/BifCl.cmake:2 (message):
  BifCl.cmake needs Zeek::BifCl
Call Stack (most recent call first):
  /opt/zeek-dev/share/zeek/cmake/ZeekPlugin.cmake:66 (include)
  CMakeLists.txt:22 (include)


-- Configuring incomplete, errors occurred!
See also "/home/awelzel/corelight-oss/zeek-community-id/build/CMakeFiles/CMakeOutput.log".
See also "/home/awelzel/corelight-oss/zeek-community-id/build/CMakeFiles/CMakeError.log".
make: *** [Makefile:401: cmake_check_build_system] Error 1
```

The workaround is to remove the `build` directory and re-run `./configure`, but IMO the 5.2.2 behavior was nicer as it just worked. Above error is also not actionable for someone running into it.

@Neverlord - I see there's some caching if bootstrapping ran and that seems to cause the issue:

https://github.com/zeek/cmake/blob/4e41cdd77f0aa617c23f37b4776a1ba5c4ea4ea3/ZeekPlugin.cmake#L59-L64

--- a/ZeekPlugin.cmake
+++ b/ZeekPlugin.cmake
@@ -6,7 +6,7 @@ function (zeek_plugin_bootstrapping)
     # Plugins that build against the source tree set ZEEK_DIST. Here, we have our
     # package file plus ZeekPluginConfig.cmake to help out.
     if (ZEEK_DIST)
-        message("USING ZEEK_DIST: ${ZEEK_DIST}")
+        message("-- Using ZEEK_DIST: ${ZEEK_DIST}")
         find_package(Zeek REQUIRED CONFIG NO_DEFAULT_PATH PATHS "${ZEEK_DIST}/build")
         return()
     endif ()
@@ -64,10 +64,8 @@ function (zeek_plugin_bootstrapping)
 endfunction ()
 
 # Make sure BifCl and BinPAC are available.
-if (NOT ZEEK_PLUGIN_INTERNAL_BUILD AND NOT ZEEK_PLUGIN_HAD_BOOTSTRAPPING)
+if (NOT ZEEK_PLUGIN_INTERNAL_BUILD)
     zeek_plugin_bootstrapping()
-    # Remember that we have called the function to not call it again.
-    set(ZEEK_PLUGIN_HAD_BOOTSTRAPPING ON CACHE BOOL "Plugin bootstrapping has completed." FORCE)
 endif ()
 
 include(BifCl)

